src/engraving/libmscore/staff.h

```diff
@@ -98,6 +98,7 @@ class Staff final : public EngravingItem
     bool _hideSystemBarLine  { false };         // no system barline if not preceded by staff with barline
     bool _mergeMatchingRests { false };         // merge matching rests in multiple voices
     HideMode _hideWhenEmpty  { HideMode::AUTO };      // hide empty staves
+    bool _isSystemObjectStaff { false };    ///< must be kept in sync with Score::systemObjectStaves
```

Comment:
Do we really need this flag here? I think we can add a method to Score that will give us the same information:

`bool Score::isSystemObjectStaff(const Staff* staff) const;
`

This way, we avoid the need for synchronization between different objects

Reply:
Yeah, the reason I introduced that flag is because I didn't know how to retrieve the property when the staff gets deleted and then re-inserted by ctrl+Z. But that's a very clumsy solution, I agree. I now realize that the right way to do it is probably to just store that info in the undo stack, as @cbjeukendrup suggested